src/sfProject.ts

```diff
@@ -602,9 +602,15 @@ export class SfProject {
    */
   public getPackageFromPath(path: string): Optional<NamedPackageDir> {
     const packageDirs = this.getPackageDirectories();
-    const match = packageDirs.find(
-      (packageDir) => basename(path) === packageDir.path || path.includes(packageDir.fullPath)
-    );
+    // resolve the given path
+    const fullPath = resolve(path);
+    const match = packageDirs.find((packageDir) => {
+      // fullPath will not have a trailing slash, so remove it from packageDir.fullPath, if it exists
+      const fullPathSansTrailingSep = packageDir.fullPath.replace(/(\\|\/)$/, '');
+      return (
+        basename(path) === packageDir.path || fullPath === fullPathSansTrailingSep || path.includes(packageDir.fullPath)
```

Comment:
if I run `sfdx force:package:beta:version:create --json -c -v user@domain.com -x -b refs/pull/46/merge -d folder_a/folder_b -f ./config/project-scratch-def.json -t Tag__20221008.7a -w 90`

then `path` will be `folder_a/folder_b`, but `packageDir.fullPath` is an absolute path so this:

```typescript
path.includes(packageDir.fullPath)
```

will always return false.

plugin-packaging:
https://github.com/salesforcecli/plugin-packaging/blob/main/src/commands/force/package/beta/version/create.ts#L187

Reply:
will replace path.includes with fullPath.includes

Reply:
@peternhale since it's possible that `getPackageFromPath` doesn't return any package, plugin-packaging should handle that case too:

https://github.com/salesforcecli/plugin-packaging/blob/main/src/commands/force/package/beta/version/create.ts#L187-#L180

TS doesn't complain because the type is assiged on line 180 but if the dir passed isn't a package sfdx will throw something like `Cannot acccess package from undefined`